modules/core/src/lib/utils/response-utils.ts

```diff
@@ -58,8 +58,8 @@ export async function makeResponse(resource: unknown): Promise<Response> {
  */
 export async function checkResponse(response: Response): Promise<void> {
   if (!response.ok) {
-    const message = await getResponseError(response);
-    throw new Error(message);
+    const error = await getResponseError(response);
+    throw error;
```

Comment:
The stack trace will be `getResponseError ... checkResponse ... <caller of checkResponse> ...` so basically one extra layer on top of whatever it was before. If it matters to you then we can move the content of `getResponseError` into this function since this function doesn't do anything else.